transaction-overview.md

```diff
@@ -155,53 +152,20 @@ rollback;
 
 In the above example, the second `insert` statement fails, and this transaction does not insert any data into the database because `rollback` is called.
 
-## Transaction sizes
-
-In TiDB, a transaction either too small or too large can impair the overall performance.
-
-### Small transactions
+## Transaction size limit
 
-TiDB uses the default autocommit setting (that is, `autocommit = 1`), which automatically issues a commit when executing each SQL statement. Therefore, each of the following three statements is treated as a transaction:
-
-```sql
-UPDATE my_table SET a = 'new_value' WHERE id = 1;
-UPDATE my_table SET a = 'newer_value' WHERE id = 2;
-UPDATE my_table SET a = 'newest_value' WHERE id = 3;
-```
+Due to the limitations of the underlying storage engine, TiDB requires a single row to be no more than 6 MB. All columns of a row are converted to bytes according to their data types and summed up to estimate the size of a single row.
 
-In this case, the latency is increased because each statement, as a transaction, uses the two-phase commit which consumes more execution time.
+TiDB supports both optimistic and pessimistic transactions, and optimistic transactions are the basis for pessimistic transactions. Because optimistic transactions first cache the changes in private memory, TiDB limits the size of a single transaction.
 
-To improve the execution efficiency, you can use an explicit transaction instead, that is, to execute the above three statements within a transaction:
+By default, TiDB sets the total size of a single transaction to no more than 100 MB. You can modify this default value via `txn-total-size-limit` in the configuration file. The maximum value of `txn-total-size-limit` is 10 GB.
 
-```sql
-START TRANSACTION;
-UPDATE my_table SET a = 'new_value' WHERE id = 1;
-UPDATE my_table SET a = 'newer_value' WHERE id = 2;
-UPDATE my_table SET a = 'newest_value' WHERE id = 3;
-COMMIT;
-```
+The actual individual transaction size limit also depends on the amount of remaining memory available to the server, because when a transaction is executed, the TiDB process consumes approximately six times more memory than the transaction size.
```

Comment:
```suggestion
The actual individual transaction size limit also depends on the amount of remaining memory available to the server, because when a transaction is executed, the TiDB process consumes approximately five times more memory than the transaction size.
```